### PR TITLE
[FIX] recurring event with datetime filter

### PR DESCRIPTION
**Pull request by mart-e**
Make the added test succeed
cf #15663 

To check : recurring all day events

### CODE REVIEW

File: addons/calendar/models/calendar.py [hunk 1]
```diff
@@ -535,19 +535,33 @@ def todate(date):
...
+            rset2 = rrule.rrulestr(str(self.rrule), dtstart=stopdate, forceset=True)
         recurring_meetings = self.search([('recurrent_id', '=', self.id), '|', ('active', '=', False), ('active', '=', True)])
 
         for meeting in recurring_meetings:
             rset1._exdate.append(todate(meeting.recurrent_id_date))
-        return [d.astimezone(pytz.UTC) for d in rset1]
+            if both:
+                rset2._exdate.append(todate(meeting.recurrent_id_date))
+        if both:
+            return zip([d.astimezone(pytz.UTC) for d in rset1], [d.astimezone(pytz.UTC) for d in rset2])
```
Comment (tde-banana-odoo) [inline addons/calendar/models/calendar.py]: No.

Comment (KangOl) [inline addons/calendar/models/calendar.py]: Different return type depending on method arguments is never a good idea.

You should have 2 methods (one can call the other)

File: addons/calendar/models/calendar.py [hunk 2]
```diff
@@ -997,16 +1013,24 @@ def get_recurrent_ids(self, domain, order=None):
                 ok = True
                 for arg in domain:
                     if str(arg[0]) in ('start', 'stop', 'final_date'):
+                        if str(arg[0]) == 'start':
+                            r_date = r_start_date
+                        else:
+                            r_date = r_stop_date
+                        if len(arg[2]) > len(r_date.strftime(DEFAULT_SERVER_DATE_FORMAT)):
```
Comment (KangOl) [inline addons/calendar/models/calendar.py]: Will fail if right part of the domain is not a string...

Comment (mart-e) [inline addons/calendar/models/calendar.py]: In which case do you get this? We are talking about recurrent events so you have a start and stop date.

Comment (KangOl) [inline addons/calendar/models/calendar.py]: example of failed domain: `[('start', '=', False)]`

Comment (mart-e) [inline addons/calendar/models/calendar.py]: nope but `('start' ,'!=', False)` did failed (and is fixed)

Comment (KangOl) [inline addons/calendar/models/calendar.py]: You will have a weird behavior with
`[('start', 'in', ['17-01-24 11:55:00', '17-01-25 11:55:00'])]`

Comment (KangOl) [inline addons/calendar/models/calendar.py]: (does it even works in standard domains?)

**Pull request merged**